### PR TITLE
fix(epg): avoid duplicate track keys for archive rows

### PR DESCRIPTION
**Pull request by 4gray**
## Summary
- Fix the EPG archive row repeater so duplicate provider start times no longer produce Angular NG0955 duplicate track-key diagnostics.
- Preserve duplicate EPG rows instead of deduping them away, so timeshift clicks remain bound to the rendered row.
- Verified PR #1094 did not introduce provider/XMLTV array merging in Xtream data-access; current fallback chooses one source at a time.

## Changes
- Add `EpgListComponent.trackProgram()` using channel/start/stop/title plus row index as the final tie-breaker.
- Use the track helper from the `@for` loop instead of `program.start`.
- Add a regression test for duplicate-start archived programs and second-row timeshift activation.

## Testing
- [x] `corepack pnpm nx test ui-epg`
- [x] `corepack pnpm nx lint ui-epg`
- [x] `corepack pnpm nx test portal-xtream-feature`
- [x] `corepack pnpm nx lint portal-xtream-feature`
- [x] `corepack pnpm nx test portal-xtream-data-access`
- [x] `corepack pnpm nx run web-e2e:e2e-ci--src/xtream.e2e.ts`

### CODE REVIEW

**Comment by 4gray** [comment]
@greptileai

File: libs/ui/epg/src/lib/epg-list/epg-list.component.ts [hunk 1]
```diff
@@ -241,6 +241,19 @@ export class EpgListComponent {
...
+        const stop = getProgramTimeMs(program.stop, program.stopTimestamp);
+
+        return [
+            program.channel ?? '',
+            Number.isFinite(start) ? start : program.start,
+            Number.isFinite(stop) ? stop : program.stop,
+            program.title ?? '',
+            index,
+        ].join('|');
+    }
```
[the remark this answers is not in the document]
Comment (4gray) [reply libs/ui/epg/src/lib/epg-list/epg-list.component.ts]: Fixed by extracting the EPG list helper functions into `epg-list.utils.ts`; `epg-list.component.ts` is now 380 lines while keeping the duplicate-start track behavior unchanged.
